### PR TITLE
uplink: change libuplink min difficulty to 0

### PR DESCRIPTION
**Pull request by jtolds**
unfortunately, making mobile clients find a hash collision of 8 zeros every time they load is just an unacceptable non starter for users of the uplink client.

we need to find a different way to deal with node id versions.

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

**Comment by bryanchriswhite** [comment]
I'm not sure what you mean about nodeID versions, perhaps you could elaborate.

Is there anywhere except for the CA where we're actually looking at the difficulty of identities because it doesn't seem like it. If that is indeed the case then this change should be fine as is but I wonder then when and where we're planning on adding that check for when we stop using the CA.

File: lib/uplink/uplink.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func NewUplink(ctx context.Context, cfg *Config) (_ *Uplink, err error) {
 	defer mon.Task()(&ctx)(&err)
 
 	ident, err := identity.NewFullIdentity(ctx, identity.NewCAOptions{
-		Difficulty:  9,
+		Difficulty:  0,
```
Comment (egonelbre) [inline lib/uplink/uplink.go]: I'm wondering why it's not breaking version checks?

Comment (jtolds) [reply lib/uplink/uplink.go]: yeah i'm confused too

**Comment by mniewrzal** [comment]
@jtolds what prevent us from merging this?

**Comment by jtolds** [comment]
anxiety. why aren't the tests failing? shouldn't they be failing? i think this is going to break everything but it didn't and i don't understand why

**Comment by jtolds** [comment]
i guess we can merge it and just find out

File: lib/uplink/uplink.go (hunk 1 above)
Comment (stefanbenten) [reply lib/uplink/uplink.go]: This is the minimum difficulty, no?

**Comment by stefanbenten** [comment]
@jtolds Isnt the specifier just the minimum difficulty it should use? It may still find one with diff 3-4 instantly. That is my observation why its not failing..

**Comment by jtolds** [comment]
there is a 50% chance that it will find a difficulty 0 hash instantly. there is a 25% chance it will find a difficulty 1 hash instantly. there is a 12.5% chance it will find a difficulty 2 hash instantly, etc. the chance of this impacting the version thing is high. this should be failing something

**Comment by stefanbenten** [comment]
Did we ever test that we ever got a Diff 0 back? I would assume that there is something in the woods, that checks for at least diff 1 :)

**Comment by jtolds** [comment]
i dunno but even that feels bad for libuplink, that we have to sit and flip coins until we get heads before we'll let the library start. that sucks

**Comment by stefanbenten** [comment]
Why flip coins? It returns the very first identity that it generated. I think there is no way to go faster than that..

**Comment by stefanbenten** [comment]
I am still thinking making the identity persist on the device is and just generate one if not being able to load/stat it.

**Comment by jtolds** [comment]
> Did we ever test that we ever got a Diff 0 back? I would assume that there is something in the woods, that checks for at least diff 1 :)
if we do this, then we're flipping coins. i agree we shouldn't flip coins

**Pull request closed without merging**